Makefile

```
@@ -24,6 +24,32 @@ LDFLAGS ?= "-X $(BUILD_DATE_VAR)=$(BUILD_DATE) -X $(BUILD_VERSION_VAR)=$(VERSION
 E2E_FLAGS ?= -installType=KindCluster
 E2E_FLAGS_DEFAULT := -test.v -ginkgo.v -ginkgo.progress -ctrRegistry $(CTR_REGISTRY) -osmImageTag $(CTR_TAG)
 
+# Reported Go version
+GO_VERSION_MAJOR = $(shell go version | grep -Po 'go\K(\d+)')
+GO_VERSION_MINOR = $(shell go version | grep -Po 'go\d+\.\K(\d+)')
+GO_VERSION_PATCH = $(shell go version | grep -Po 'go\d+\.\d+\.\K(\d+)')
```

Builtin `grep` on macOS doesn't support `-P`. I'm looking for an alternative.

---

This works on macOS and linux AFAICT:
```suggestion
GO_VERSION_MAJOR = $(shell go version | sed -E 's/.*go([[:digit:]]+)\..*/\1/')
GO_VERSION_MINOR = $(shell go version | sed -E 's/.*go[[:digit:]]+\.([[:digit:]]+)\..*/\1/')
GO_VERSION_PATCH = $(shell go version | sed -E 's/.*go[[:digit:]]+\.[[:digit:]]+\.([[:digit:]]+)[[:space:]].*/\1/')
```